sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1047,8 +1048,8 @@ case class ScalaUDF(
 
     // The type converters for inputs and the result.
     val converters: Array[Any => Any] = children.zipWithIndex.map { case (c, i) =>
-      createToScalaConverter(i, c.dataType)
-    }.toArray :+ CatalystTypeConverters.createToCatalystConverter(dataType)
+      scalaConverter(i, c.dataType)
+    }.toArray :+ createToCatalystConverter(dataType)
```

Comment:
does the UDF return type support case class?

Reply:
yes.